src/drawer.tsx

```diff
@@ -0,0 +1,307 @@
+// https://www.figma.com/design/NfM16IJ4ffPVEdiFbtU0Lu/%F0%9F%94%B8-Drawer-Specs?node-id=10-5397&t=58YG59t526tkk7dP-1
+'use client';
+import * as React from 'react';
+import Stack from './stack';
+import {Text3, Text4, Text5} from './text';
+import {vars} from './skins/skin-contract.css';
+import {IconButton} from './icon-button';
+import IconCloseRegular from './generated/mistica-icons/icon-close-regular';
+import Box from './box';
+import * as styles from './drawer.css';
+import classnames from 'classnames';
+import {Portal} from './portal';
+import {useIsInViewport, useScreenSize, useTheme} from './hooks';
+import FocusTrap from './focus-trap';
+import {useSetModalStateEffect} from './modal-context-provider';
+import ButtonLayout from './button-layout';
+import {ButtonLink, ButtonPrimary, ButtonSecondary} from './button';
+import Divider from './divider';
+import {getPrefixedDataAttributes} from './utils/dom';
+import * as tokens from './text-tokens';
+
+import type {DataAttributes, HeadingType, TrackingEvent} from './utils/types';
+
+const PADDING_X_DESKTOP = 40;
+const PADDING_X_MOBILE = 16;
+const PADDING_X_TABLET = 16;
```

Comment:
Update to 32 to match the spacings in tablet, updated in specs also:

https://www.figma.com/design/NfM16IJ4ffPVEdiFbtU0Lu/%F0%9F%94%B8-%5BREADY%5D-Drawer-Specs?node-id=10-5397&t=ODR4pg1TRGAgvsF8-1

Reply:
done